triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerLogContext.scala

```diff
@@ -51,24 +51,44 @@ final class TriggerLogContext private (
 
   import ToLoggingContext._
 
+  private[this] var callback: (String, TriggerLogContext) => Unit = (_, _) => ()
+
+  private[trigger] def setCallback(callback: (String, TriggerLogContext) => Unit): Unit = {
+    this.callback = callback
+  }
```

Comment:
Use of a callback allows the calling context to decide how trigger log context data may be used (e.g. to emit via a metrics agent or for use by testing)